src/main/java/net/minecraftforge/items/pickup/ItemPickupReasons.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.items.pickup;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.vehicle.MinecartHopper;
+import net.minecraft.world.level.block.entity.HopperBlockEntity;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Class containing all standard ItemPickupReasons provided by Forge.
+ * <p>Make use of {@link #getOrCreate(String)} for building new {@link ItemPickupReason} instances.
+ */
+@ApiStatus.NonExtendable
+public final class ItemPickupReasons
+{
+    /**
+     * Player is attempting to pick up an item
+     * <p>{@code collector} types should result in the {@link Player} attempting to pick up the item.
+     */
+    public static final ItemPickupReason PLAYER = getOrCreate("player");
+
+    /**
+     * LivingEntity is attempting to pick up an item
+     * <p>{@code collector} types should result in the {@link LivingEntity} attempting to pick up the item.
+     */
+    public static final ItemPickupReason LIVING_ENTITY = getOrCreate("living_entity");
+
+    /**
+     * Hopper is attempting to suck in an item
+     * <p>{@code collector} types should result in the {@link HopperBlockEntity} or {@link MinecartHopper} {@link Container containers} attempting to pick up the item.
+     */
+    public static final ItemPickupReason HOPPER = getOrCreate("hopper");
+
+    /**
+     * {@return all registered pickup reasons}
+     */
+    public static Collection<ItemPickupReason> all()
+    {
+        return Collections.unmodifiableCollection(ItemPickupReason.reasons.values());
```

Comment:
Why would anyone need access to all values?

Reply:
people shouldn't need to access all values but I initially copied `ToolAction` which too has a method to obtain all registered values, could simply remove this method.
`ToolAction#getActions`

This also goes for `ItemPickupReason` not being a record, while it could be a record, I initially copied `ToolAction` which is not a record.
[_but now that I think of it, that may be because it was created before records were a thing?_]